src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -207,6 +207,19 @@ def validate_cert(namespace):
             pass
 
 
+def validate_delegations(cmd, namespace):
+    if namespace.delegations:
+        Delegation = cmd.get_models('Delegation')
+        delegations = []
+        for i, item in enumerate(namespace.delegations):
+            if '/' not in item and len(item.split('.')) == 3:
```

Comment:
We're allowing for "Microsoft.Sql.servers"? Do other commands do this?

Reply:
No, this is something very specific to this command.  When you run the command to get the values it looks like:
```
"name": "Microsoft.Sql.servers",
"serviceName": "Microsoft.Sql/servers"
```

The `--delegations` property requires serviceName, so I do this conversion because I know there is a 100% chance that someone will use name instead and post an issue when the service doesn't accept it.

Reply:
@tjprescott is it possible to add a transformer to edit the results to look like the serviceName?